tools/base/utils.py

```diff
@@ -0,0 +1,28 @@
+#
+# Provides shared utils used by other python modules
+#
+
+import os
+import tempfile
+from configparser import ConfigParser
+from contextlib import contextmanager
+from typing import Iterator
+
+
+# this is testing specific - consider moving to tools.testing.utils
+@contextmanager
+def coverage_with_data_file(data_file: str) -> Iterator[str]:
+    """This context manager takes the path of a data file
+    and creates a custom coveragerc with the data file path included.
+
+    The context is yielded the path to the custom rc file.
+    """
+    parser = ConfigParser()
+    parser.read(".coveragerc")
+    parser["run"]["data_file"] = data_file
+    # use a temporary .coveragerc
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tmprc = os.path.join(tmpdir, ".coveragerc")
```

Comment:
Probably easier to write this whole block with `pathlib` `write_text`.

Reply:
it wants to be a tempdir so im not clear how that would help

Reply:
oh i see not using open

Reply:
not really - its using `parser.write`